src/pages/reset-password/reset-password.ts

```diff
@@ -133,47 +135,35 @@ export class ResetPasswordPage implements OnInit {
               this.cacheService.setLocal('timelineID', data.Timelines[0].Timeline.id);
               this.cacheService.setLocal('teams', data.Teams);
               this.cacheService.setLocal('gotNewItems', false);
-              // get game_id data after login
-              this.gameService.getGames()
-                  .subscribe(
-                    data => {
-                      console.log("game data: ", data);
-                      _.map(data, (element) => {
-                        console.log("game id: ", element[0].id);
-                        this.cacheService.setLocal('game_id', element[0].id);
-                      });
-                    },
-                    err => {
-                      console.log("game err: ", err);
-                    }
-                  );
-              // get milestone data after login
-              this.authService.getUser()
-                  .subscribe(
-                    data => {
-                      this.cacheService.setLocal('name', data.User.name);
-                      this.cacheService.setLocal('email', data.User.email);
-                      this.cacheService.setLocal('program_id', data.User.program_id);
-                      this.cacheService.setLocal('project_id', data.User.project_id);
-                    },
-                    err => {
-                      console.log(err);
-                    }
-                  );
-              // get milestone data after login
-              this.milestoneService.getMilestones()
-                  .subscribe(
-                    data => {
-                      loading.dismiss().then(() => {
-                        console.log(data[0].id);
-                        this.milestone_id = data[0].id;
-                        this.cacheService.setLocal('milestone_id', data[0].id);
-                        console.log("milestone id: " + data[0].id);
-                        loading.dismiss();
-                        this.navCtrl.push(TabsPage).then(() => {
-                          this.viewCtrl.dismiss(); // close the login modal and go to dashaboard page
-                          window.history.replaceState({}, '', window.location.origin);
-                        });
+              let getGame = this.gameService.getGames();
+              let getUser = this.authService.getUser();
+              let getMilestone = this.milestoneService.getMilestones();
+              Observable.forkJoin([getGame, getUser, getMilestone])
+                .subscribe(
+                  results => {
+                    loading.dismiss().then(() => {
+                      // results[0] game API data
+                      this.gameID = results[0].Games[0].id;
```

Comment:
@DamengRandom same code repeated [[*login.ts*](https://github.com/intersective/practera-sdk-ionic/pull/45/files#r149849451), [*magic-link.ts*](https://github.com/intersective/practera-sdk-ionic/pull/45/files#r149850469), [*register.page.ts*](https://github.com/intersective/practera-sdk-ionic/pull/45/files#r149850852)]

Reply:
Yup sure, I will refactor it, thx @trtshen ^_^